### ISSUE TITLE
[clang-tidy] cppcoreguidelines-virtual-class-destructor prints misleading diagnostics

### ISSUE THREAD

**Issue by Sir-Walrus** [llvm/llvm-project#224686]
```c++
class a {














protected:
    virtual ~a() = default;
};
```
--checks=cppcoreguidelines-virtual-class-destructor for clang-tidy, nothing in particular needed for Clang. Linebreaks necessary. https://godbolt.org/z/xaYxEMvMb

Output:
```
<source>:1:7: warning: destructor of 'a' is protected and virtual [cppcoreguidelines-virtual-class-destructor]
    1 | class a {
      |       ^
<source>:1:7: note: make it protected and non-virtual
    1 | class a {
      |       ^
```
Expected: Point to the dtor, and/or the word `protected`. Pointing to the class head makes me think the dtor is defaulted, and the dtor I added ended up as part of a different class or something.
